google-cloud-storage/src/main/scala/akka/stream/alpakka/googlecloud/storage/javadsl/GCStorage.scala

```diff
@@ -233,6 +234,38 @@ object GCStorage {
                    contentType: ContentType): Source[StorageObject, NotUsed] =
     GCStorageStream.putObject(bucket, objectName, data.asScala, contentType.asInstanceOf[ScalaContentType]).asJava
 
+  /**
+   * Uploads object by making multiple requests
+   *
+   * @see https://cloud.google.com/storage/docs/json_api/v1/how-tos/resumable-upload
+   *
+   * @param bucket the bucket name
+   * @param objectName the object name
+   * @param contentType `ContentType`
+   * @param chunkSize the size of the request sent to google cloud storage in bytes, must be a multiple of 256KB
+   * @param metadata custom metadata for the object
+   * @return a `Sink` that accepts `ByteString`'s and materializes to a `Future` of `StorageObject`
+   */
+  def resumableUpload(bucket: String,
+                      objectName: String,
+                      contentType: ContentType,
+                      chunkSize: java.lang.Integer,
+                      metadata: java.util.Map[String, String]): Sink[ByteString, CompletionStage[StorageObject]] = {
+    assert(
+      (chunkSize >= (256 * 1024)) && (chunkSize % (256 * 1024) == 0),
+      "Chunk size must be a multiple of 256KB"
+    )
```

Comment:
Make the old overloads call this one (with a `Map.empty`) so we can DRY up the pre-conditions.

Reply:
Okay so I just had a look at this, and there's a tiny issue with doing this:
Currently the overload takes a util.Map, and converts that to Some[Map[String, String]]. It's currently the option on which we decide to create a request with metadata or not. Passing it through that Some, means we create a "metadata request", but with no metadata. That leaves us with a few different options:
- Do we make the file upload mock expect this, and send requests with an empty metadata map?
- Do we check both the Option being defined, *and* the metadata being nonEmpty (at which point the Option is a bit redundant, and we could just use an empty map for no metadata)
- Do we simply extract the chunk size preconditions so they aren't repeated

Not sure I have a huge preference, maybe using the empty map case as I did debate the merit of using an option in the first place?

Reply:
Ah, makes sense. Using the `Option` seems fine.

> Do we simply extract the chunk size preconditions so they aren't repeated, either via another (private?) overload that takes an optional metadata map, or just a private method for the chunk size?

I guess it's not a big deal to DRY things up, but this suggestion is a reasonable solution.